docs/experiments-fid-reader.md

```diff
@@ -0,0 +1,168 @@
+# PyGaggle: Fusion-in-Decoder (FiD) Baselines
+
+This page contains instructions for running the FiD tools.
+
+It's recommended to run the experiment in virtualenv (with Python-version=3.7).
+
+## Natural Questions (NQ)
+End-to-end answer prediction using **Hybrid Retrieval**
+
+Run nvidia-smi to make sure gpu is set up.
+```bash
+pip install pyserini
+export JAVA_HOME="/usr/lib/jvm/java-11-openjdk-amd64"
+git clone https://github.com/castorini/pygaggle.git 
+cd pygaggle
+pip install --editable .
+pip install -r requirements.txt
+gsutil cp gs://ron-random/runs/run.encoded.dkrr.test.json data/
+```
+
+Now download the model accordingly, and the model name can be found in FiD/get-model.sh. (i.e. nq_reader_base)
+
+```bash
+git clone https://github.com/facebookresearch/FiD.git
+cd FiD
+bash get-model.sh -m nq_reader_base
+pip install -r requirements.txt
+!pip install transformers==4.10.0
+```
+
+After downloading the model, don't forget to change the model name being used in pygaggle/qa/fid_reader.py
+
+Then we run the inference and evaluation in the TOP level directory of Pygaggle:
+```bash
+$ python -um pygaggle.run.evaluate_fid_ranker --task wikipedia --retriever score --reader fid \
```

Comment:
where is the nq_reader_base passed to the script, I'm confused?

Reply:
nq_reader_base needs to be changed in fid_reader.py

Reply:
I'll update that too